### PR TITLE
LMN-1036 Add image gallery carousel

### PR DESCRIPTION
**Pull request by marianeum**
First part of the image gallery component - the carousel which is the entry point.

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/carousel/BpkCarousel.kt [hunk 1]
```diff
@@ -30,12 +30,27 @@ import androidx.compose.ui.platform.testTag
...
+        state = state,
+        modifier = modifier,
+        content = content,
+        overlayContent = { pageIndicator -> Box(Modifier.align(Alignment.BottomCenter)) { pageIndicator?.invoke() } },
+    )
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+internal fun BpkCarousel(
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/carousel/BpkCarousel.kt:49]: created an internal reusable function - the gallery carousel has different spacing + more content for the overlay

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGallery.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+fun BpkImageGalleryCarousel(
+    state: BpkCarouselState,
+    modifier: Modifier = Modifier,
+    onImageClicked: ((Int) -> Unit)? = null,
+    content: @Composable (BoxScope.(Int) -> Unit),
+) {
+    val interactionSource = remember { MutableInteractionSource() }
+
+    BpkCarousel(
+        modifier = modifier.clickable(interactionSource = interactionSource, indication = null) {
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGallery.kt:48]: the click interaction conflicts with the scroll interaction - disabling the ripple effect

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGallery.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                modifier = Modifier
+                    .align(Alignment.BottomCenter)
+                    .fillMaxWidth()
+                    .padding(
+                        horizontal = BpkSpacing.Base,
+                        vertical = ImageGalleryPreviewVerticalSpacing,
+                    ),
+            ) {
+                pageIndicator?.invoke()
+                BpkBadge(
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/imagegallery/BpkImageGallery.kt:66]: adding the page count badge as well

File: docs/view/Dialog/screenshots/flare.png
Comment (marianeum) [inline docs/view/Dialog/screenshots/flare.png:1]: turns out i missed updating this one last time when i removed picasso

**Review by alejandrorosas — APPROVED** [review]

**Pull request merged**